### PR TITLE
Consider locales encoding

### PR DESCRIPTION
**Pull request by imobachgs**
## Problem

See #987.

## Solution

* The encoding should be included in the locale, although at this point we always use `UTF-8` (check `localectl list-locales`).
* I adapted the web UI to ignore the encoding (it is only interested in the language).

## Updating the spec file

I updated the `spec` file according to the latest changes of cargo_vendor:

```
* cargo_config is no longer created - it's part of the vendor.tar now
    * You can safely remove lines related to cargo_config from your spec file

* multiple cargotoml files can be specified and share a single vendor.tar
    * If multiple cargo.toml files are present update does not work. This is a known
      limitation of the process

* cargo_audit is now part of cargo_vendor, meaning you don't have to configure it separately
```

## Testing

* Manually tested using a branch in OBS.

### CODE REVIEW

**Review by dgdavid — COMMENTED** [review]
Draft: So far, it looks good to me :+1:

**Review by dgdavid — APPROVED** [review]
LGTM 👍 However, you might want a review from another colleague for the Rust part. Up to you 😉

**Comment by imobachgs** [comment]
> LGTM 👍 However, you might want a review from another colleague for the Rust part. Up to you 😉

Thanks a lot! I just wanted @mvidner to, please, have a look as our language expert :-)

File: rust/agama-dbus-server/src/l10n/helpers.rs [hunk 1]
```diff
@@ -23,8 +23,7 @@ pub fn init_locale() -> Result<LocaleCode, Box<dyn std::error::Error>> {
 ///
 pub fn set_service_locale(locale: &LocaleCode) {
     // Let's force the encoding to be 'UTF-8'.
```
Comment (joseivanlopez) [inline rust/agama-dbus-server/src/l10n/helpers.rs:25]: This comment does not apply anymore, right?

File: service/lib/agama/ui_locale.rb [hunk 2]
```diff
@@ -46,10 +46,11 @@ def initialize(locale_client, &block)
 
     def change_locale(locale)
       # TODO: check if we can use UTF-8 everywhere including strange arch consoles
```
Comment (joseivanlopez) [inline service/lib/agama/ui_locale.rb:48]: And this TODO?

**Review by joseivanlopez — APPROVED** [review]

**Pull request merged**